repository/src/main/java/org/alfresco/repo/domain/node/AbstractNodeDAOImpl.java

```diff
@@ -5035,6 +5037,8 @@ protected abstract void selectChildAssocsWithoutParentAssocsOfType(
             Long parentNodeId,
             QName assocTypeQName,
             ChildAssocRefQueryCallback resultsCallback);
+    protected abstract List<String> selectRootGroupsNames(
+            Long parentNodeId);

```

Comment:
I don't get why we have an overridden method that just passes through to an abstract method.

Reply:
I was following the same approach of the ChildAssocsOfParentWithoutParentAssocsOfType (the original method that was being called) and didn't notice that for the new method this wasn't needed.